src/fides/api/alembic/migrations/versions/c9abc9bbaa49_add_custom_report_table.py

```diff
@@ -0,0 +1,51 @@
+"""add custom report table
+
+Revision ID: c9abc9bbaa49
+Revises: 75bb9ee843f5
+Create Date: 2024-09-26 23:19:58.766385
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "c9abc9bbaa49"
+down_revision = "75bb9ee843f5"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table(
+        "plus_custom_report",
+        sa.Column("id", sa.String(length=255), nullable=False),
+        sa.Column(
+            "created_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column(
+            "updated_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column("name", sa.String(), nullable=True),
+        sa.Column("type", sa.String(), nullable=False),
+        sa.Column("created_by", sa.String(), nullable=True),
+        sa.Column("config", postgresql.JSONB(astext_type=sa.Text()), nullable=True),
```

Comment:
Are we storing the `table_state` and `column_map` within the `config` (as opposed to using two separate columns) because we expect that we'll need more things in the config if/when we implement custom reports in other parts of the UI?

Reply:
That's right, especially for `table_state`, Jason is still trying to figure out the schema for the front-end. It's tricky because the metadata to capture a Tanstack table is different from an ANT table so I just provided a flexible solution for now.